### PR TITLE
Fixed: Reduce parameters required to add a new series

### LINKED ISSUE #1403: Add Series: Default properties that are NOT NULL in the database
For adding a series we have a limited number of properties that are required, but we have a larger number of fields that are required to successfully save it to the data base (`NOT NULL` DB columns), we should default anything we expect to not be null, but don't care if its been set or remove the `NOT NULL` restriction in the DB (and make sure they are being deserialized properly).

We can default it in the API resource, the model or directly in the DB.

### PR DESCRIPTION
**Pull request by markus101**
#### Database Migration
NO

#### Description
Removes the requirement to send the full JSON body (from `/series/lookup`) when adding a new series and requires only the basic information (so Sonarr knows where to put it and what quality to get it in).

#### Issues Fixed or Closed by this PR

Fixes #1403

### CODE REVIEW

File: src/NzbDrone.Api/Series/SeasonResource.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public static List<SeasonResource> ToResource(this IEnumerable<Season> models)
 
         public static List<Season> ToModel(this IEnumerable<SeasonResource> resources)
         {
-            return resources.Select(ToModel).ToList();
+            return resources?.Select(ToModel).ToList();
```
Comment (Taloth) [inline src/NzbDrone.Api/Series/SeasonResource.cs]: I'd prefer a separate if like with the other resource converters.

File: src/NzbDrone.Core.Test/TvTests/AddSeriesFixture.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                  .Returns(new ValidationResult(new List<ValidationFailure>
+                                                {
+                                                    new ValidationFailure("Path", "Test validation failure")
+                                                }));
+
+            Assert.Throws<ValidationException>(() => Subject.AddSeries(newSeries));
+
+            ExceptionVerification.ExpectedErrors(1);
+        }
+    }
```
Comment (Taloth) [inline src/NzbDrone.Core.Test/TvTests/AddSeriesFixture.cs]: Maybe add test to check for duplicate slug while we're at it?

File: src/NzbDrone.Core/Tv/AddSeriesService.cs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            newSeries.CleanTitle = newSeries.Title.CleanSeriesTitle();
+            newSeries.SortTitle = SeriesTitleNormalizer.Normalize(newSeries.Title, newSeries.TvdbId);
+            newSeries.Added = DateTime.UtcNow;
+
+            _seriesService.AddSeries(newSeries);
+
+            return newSeries;
+        }
+
+        private Series SeriesGetSeriesInfo(Series newSeries)
```
Comment (Taloth) [inline src/NzbDrone.Core/Tv/AddSeriesService.cs]: Just `GetSeriesInfo`

File: src/NzbDrone.Core/Tv/AddSeriesService.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+            var series = tuple.Item1;
+
+            series.Path = newSeries.Path;
+            series.RootFolderPath = newSeries.RootFolderPath;
+            series.Monitored = newSeries.Monitored;
+            series.AddOptions = newSeries.AddOptions;
+            series.ProfileId = newSeries.ProfileId;
+            series.SeasonFolder = newSeries.SeasonFolder;
+            series.SeriesType = newSeries.SeriesType;
```
Comment (Taloth) [inline src/NzbDrone.Core/Tv/AddSeriesService.cs]: Do all these properties have sensible defaults? Imagine someone adding a series without `{ ..., "monitored": true }`.

Also, it might be better to extract setting these properties to a separate method, or even class.
iirc we have some code that defines what the user is allowed to change on a series. It's a powerplay in SeriesModule between seriesService.UpdateService and first having to fetch the current state from the seriesService. Basically, SeriesResource.ToModel only applies properties the user is allowed to change. (https://github.com/Sonarr/Sonarr/blob/816cf608fcef07e375aa750a863d0a7365108981/src/NzbDrone.Api/Series/SeriesResource.cs#L208-L225)
So having the same list of 'mutable' properties on two locations isn't very maintainable.

File: src/NzbDrone.Api/Series/SeasonResource.cs (hunk 1 above)
Comment (markus101) [inline src/NzbDrone.Api/Series/SeasonResource.cs]: 👍 This was me testing the magic (and the waters).

Comment (Taloth) [inline src/NzbDrone.Api/Series/SeasonResource.cs]: I'm not opposed to the ?. operator in general, but in this case having a separate `if` makes it more clear that it can accept and emit a null.

File: src/NzbDrone.Core/Tv/AddSeriesService.cs (hunk 4 above)
Comment (markus101) [inline src/NzbDrone.Core/Tv/AddSeriesService.cs]: We allow the TVDB ID to be changed via the API? That seems not great, also seasons makes sense (for monitoring), which I forgot about (though we'll want to make sure they were actually set, not null or empty).

File: src/NzbDrone.Api/Series/SeasonResource.cs (hunk 1 above)
Comment (markus101) [inline src/NzbDrone.Api/Series/SeasonResource.cs]: Yeah, returning null for a list is bad form anyways.

File: src/NzbDrone.Core/Tv/AddSeriesService.cs (hunk 3 above)
Comment (kayone) [inline src/NzbDrone.Core/Tv/AddSeriesService.cs]: how about `MergeSeriesInfo` or `GetCombinedSeriesInfo`

File: src/NzbDrone.Core/Tv/AddSeriesService.cs [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+            newSeries = SeriesGetSeriesInfo(newSeries);
+
+            if (string.IsNullOrWhiteSpace(newSeries.Path))
+            {
+                var folderName = _fileNameBuilder.GetSeriesFolder(newSeries);
+                newSeries.Path = Path.Combine(newSeries.RootFolderPath, folderName);
+            }
+
+            var validationResult =_addSeriesValidator.Validate(newSeries);
```
Comment (kayone) [inline src/NzbDrone.Core/Tv/AddSeriesService.cs]: I think this should be done in `SeriesService.AddSeries`

File: src/NzbDrone.Api/Series/SeasonResource.cs (hunk 1 above)
Comment (kayone) [inline src/NzbDrone.Api/Series/SeasonResource.cs]: +1 for `?`
-1 for returning null.
